feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/presentation/nft/send/confirm/ConfirmNftSendViewModel.kt

```diff
@@ -0,0 +1,194 @@
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.FeeLoaderMixin
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.launch
+
+class ConfirmNftSendViewModel(
+    private val nftSendInteractor: NftSendInteractor,
+    private val router: NftRouter,
+    private val addressIconGenerator: AddressIconGenerator,
+    private val externalActions: ExternalActions.Presentation,
+    private val chainRegistry: ChainRegistry,
+    private val selectedAccountUseCase: SelectedAccountUseCase,
+    private val addressDisplayUseCase: AddressDisplayUseCase,
+    private val validationExecutor: ValidationExecutor,
+    private val walletUiUseCase: WalletUiUseCase,
+    feeLoaderMixinFactory: FeeLoaderMixin.Factory,
+    val transferDraft: NftTransferDraft,
+    private val resourceManager: ResourceManager
+) : BaseViewModel(),
+    ExternalActions by externalActions,
+    Validatable by validationExecutor {
+
+    private val originChain by lazyAsync { chainRegistry.getChain(transferDraft.chainId) }
+
+    private val commissionAssetFlow = nftSendInteractor.commissionAssetFlow(transferDraft.chainId)
+        .inBackground()
+        .share()
+
+    private val commissionTokenFlow = nftSendInteractor.commissionAssetFlow(transferDraft.chainId)
+        .map { it.token }
+        .inBackground()
+        .share()
+
+    val originFeeMixin: FeeLoaderMixin.Presentation = feeLoaderMixinFactory.create(commissionTokenFlow)
+
+    private val currentAccount = selectedAccountUseCase.selectedMetaAccountFlow()
+        .inBackground()
+        .share()
+
+    val recipientModel = flowOf {
+        createAddressModel(
+            address = transferDraft.recipientAddress,
+            chain = originChain(),
+            resolveName = true
+        )
+    }
+        .inBackground()
+        .share()
+
+    val senderModel = currentAccount.mapLatest { metaAccount ->
+        createAddressModel(
+            address = metaAccount.requireAddressIn(originChain()),
+            chain = originChain(),
+            resolveName = false
+        )
+    }
+        .inBackground()
+        .share()
+
+    val originChainUi = flowOf { originChain() }
+        .map(::mapChainToUi)
+        .shareInBackground()
+
+    val wallet = walletUiUseCase.selectedWalletUiFlow()
+        .inBackground()
+        .share()
+
+    private val _transferSubmittingLiveData = MutableStateFlow(false)
+
+    val sendButtonStateLiveData = _transferSubmittingLiveData.map { submitting ->
+        if (submitting) {
+            ButtonState.PROGRESS
+        } else {
+            ButtonState.NORMAL
+        }
+    }
```

Comment:
We have shortcut in PrimaryButton `setProgress(boolean)` so this flow is not needed, you can use something like

`val showButtonProgress: Flow<Boolean> = _transferSubmittingFlow`

Reply:
Renamed